### PR TITLE
Forbid commands in DMs with in_whitelist_check

### PR DESCRIPTION
**Pull request by Akarys42**
This will prevent most commands from running in DMs, which would break most commands and lead to moderation issues.

### CODE REVIEW

File: bot/utils/checks.py [hunk 1]
```diff
@@ -39,6 +39,7 @@ def in_whitelist_check(
     channels: Container[int] = (),
     categories: Container[int] = (),
     roles: Container[int] = (),
+    allow_dms: bool = False,
```
Comment (ToxicKidz) [inline bot/utils/checks.py:42]: I don't think this parameter was used anywhere.

**Review by ToxicKidz — CHANGES_REQUESTED** [review]

File: bot/utils/checks.py (hunk 1 above)
Comment (Akarys42) [reply bot/utils/checks.py:42]: *play Jedi tricks* you didn't see anything

File: bot/utils/checks.py [hunk 2]
```diff
@@ -66,21 +67,23 @@ def in_whitelist_check(
...
 
-    if channels and ctx.channel.id in channels:
-        log.trace(f"{ctx.author} may use the `{ctx.command.name}` command as they are in a whitelisted channel.")
-        return True
-
-    # Only check the category id if we have a category whitelist and the channel has a `category_id`
-    if categories and hasattr(ctx.channel, "category_id") and ctx.channel.category_id in categories:
-        log.trace(f"{ctx.author} may use the `{ctx.command.name}` command as they are in a whitelisted category.")
-        return True
+    if allow_dms or ctx.guild is not None:
```
Comment (ChrisLovering) [inline bot/utils/checks.py:70]: Do we need `is not None` here? Could it just be `or ctx.guild`?

**Comment by Akarys42** [comment]
Due to our (probably too deeply) nested checks, these changes shouldn't be solving any of our issues according to sir @HassanAbouelela. Closing this PR.

**Pull request closed without merging**